src/Persistent/Pure.hs

```diff
@@ -220,94 +233,94 @@ type AddDb m a = UserWithProto a -> PersistM m => m a
 -- It could make sense for the traversal to point to more than one target for instance
 -- to index the record at different locations. For instance we could keep an additional
 -- global map of the comments, votes, likes and still call @addDb@ only once.
-addDb :: (HasMetaInfo a, FromProto a) => AulaTraversal (AMap a) -> AddDb m a
+addDb :: forall a. (HasMetaInfo a, FromProto a) => AulaTraversal (AMap a) -> AddDb a
 addDb l (cUser, pa) = do
-    assertPersistM $ do
-        db <- getDb id
-        let len = lengthOf l db
+    assertAulaDataM $ do
+        len <- lengthOf l <$> askDb id
         when (len /= 1) $ do
             fail $ "Persistent.Api.addDb expects the location (lens, traversal) "
                 <> "to target exactly 1 field not " <> show len
-    a <- fromProto pa <$> nextMetaInfo cUser
-    modifyDb l $ at (a ^. _Id) .~ Just a
+    -- FIXME: reduce code: call 'addDbValue' instead of the following three lines.  (this is another lens puzzle)
+    a :: a <- fromProto pa <$> nextMetaInfo cUser
+    modifyDb_ l $ at (a ^. _Id) .~ Just a
```

Comment:
Actually `addDbValue` needs to be changed for that. Let's keep it like that for now.